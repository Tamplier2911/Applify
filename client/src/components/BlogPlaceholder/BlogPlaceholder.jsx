// import "./BlogPlaceholder.scss";
import React from "react";

// JS Rendering CSS
import {
  BlogPlaceholderContainer,
  BlogPlaceholderWrapper,
  BlogPlaceholderSVG
} from "./BlogPlaceholderStyles";

const BlogPlaceholder = ({ pos }) => {
  const big = [0, 3, 4];
  const isBig = big.includes(pos) ? 1 : 0;
  return (
    <BlogPlaceholderContainer big={isBig}>
      <BlogPlaceholderWrapper>
        <BlogPlaceholderSVG />
      </BlogPlaceholderWrapper>
    </BlogPlaceholderContainer>
  );
};

export default BlogPlaceholder;
