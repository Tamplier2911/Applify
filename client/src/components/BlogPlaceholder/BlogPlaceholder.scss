/*

.blog-placeholder {
  display: grid;

  &__wrap {
    display: grid;
    align-items: center;
    justify-items: center;

    &:hover .blog-placeholder__svg {
      fill: var(--cl-primary);
    }
  }

  &__svg {
    width: 4rem;
    height: 4rem;
    transition: fill 0.3s;
  }
}

*/
