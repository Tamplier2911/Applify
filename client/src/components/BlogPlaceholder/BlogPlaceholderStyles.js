import styled, { css } from "styled-components";

// svg
import { ReactComponent as BlogSVG } from "../../assets/svg/adminpage-blogs.svg";

const blogFadeInAnimation = css`
  @keyframes blogFadeIn {
    0% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
`;

const bigOrSmall = props => {
  const { big } = props;
  if (big) {
    return `grid-column: span 2;
              @media only screen and (max-width: 425px) {
                  grid-column: 1 / 2;
              }`;
  } else {
    return `grid-column: span 1;
              @media only screen and (max-width: 768px) {
                  grid-column: span 2;
              }
              @media only screen and (max-width: 425px) {
                  grid-column: 1 / 2;
              }`;
  }
};

export const BlogPlaceholderContainer = styled.div`
  ${bigOrSmall}
  ${blogFadeInAnimation}
  cursor: pointer;
  display: grid;

  font-family: var(--font-title);
  font-weight: 300;
  color: var(--cl-white);

  min-height: 20rem; // double check
  max-height: 20rem; // double check
  box-shadow: 0rem 0rem 0.5rem var(--cl-square-shadow);

  transition: filter 0.5s, box-shadow 0.3s;
  animation: 1s blogFadeIn;
  filter: brightness(90%);

  @media only screen and (max-width: 425px) {
    grid-template-rows: repeat(10, 1fr);
    max-height: 30rem;
  }

  &:hover {
    filter: brightness(130%);
    box-shadow: 0rem 0rem 1rem var(--cl-primary);
  }
`;

export const BlogPlaceholderSVG = styled(BlogSVG)`
  width: 4rem;
  height: 4rem;
  fill: var(--cl-font);
  transition: fill 0.3s;
`;

export const BlogPlaceholderWrapper = styled.div`
  grid-row: 1 / -1;
  display: grid;
  align-items: center;
  justify-items: center;

  &:hover ${BlogPlaceholderSVG} {
    fill: var(--cl-primary);
  }
`;
