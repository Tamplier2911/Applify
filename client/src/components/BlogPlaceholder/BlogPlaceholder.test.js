import React from "react";

// component
import BlogPlaceholder from "./BlogPlaceholder";

// utils
import { setShallow, findByTestAttr, checkProps } from "../../tests/testUtils";

describe("<BlogPlaceholder />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setShallow(<BlogPlaceholder />);
  });

  it("renders without an error", () => {
    const blogPlaceholder = findByTestAttr(wrapper, "blog-placeholder");
    expect(blogPlaceholder.length).toBe(1);
  });

  it("renders wrapper children", () => {
    const blogPlaceholderWrap = findByTestAttr(
      wrapper,
      "blog-placeholder-wrap"
    );
    expect(blogPlaceholderWrap.length).toBe(1);
  });

  it("recieves correct props", () => {
    const expectedProps = { pos: 1 };
    const err = checkProps(BlogPlaceholder, expectedProps, "pos");
    expect(err).toBeUndefined();
  });

  it("matches snapshot", () => {
    const blogPlaceholder = findByTestAttr(wrapper, "blog-placeholder");
    expect(blogPlaceholder).toMatchSnapshot();
  });
});
