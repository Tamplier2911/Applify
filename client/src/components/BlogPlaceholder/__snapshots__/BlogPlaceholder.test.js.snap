// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPlaceholder /> matches snapshot 1`] = `
<styled.div
  big={0}
  data-test="blog-placeholder"
>
  <styled.div
    data-test="blog-placeholder-wrap"
  >
    <Styled(Component) />
  </styled.div>
</styled.div>
`;
